workflow/snakemake_rules/common.smk

```diff
@@ -145,15 +145,37 @@ def _get_max_date_for_frequencies(wildcards):
             date.today() - offset
         )
 
-def _get_first(config, *keys):
-    """
-    Get the value of the first key in *keys* that exists in *config* and has a
-    non-empty value.
-
-    Raises a :class:`KeyError` if none of the *keys* exist with a suitable
-    value.
-    """
-    for key in keys:
-        if config.get(key) not in {"", None}:
-            return config[key]
-    raise KeyError(str(keys))
+def _get_upload_inputs(wildcards):
+    # The main workflow supports multiple inputs/origins, but our desired file
+    # structure under data.nextstrain.org/files/ncov/open/… is designed around
+    # a single input/origin.  Intermediates (aligned, masked, filtered, etc)
+    # are specific to each input/origin and thus do not match our desired
+    # structure, while builds (global, europe, africa, etc) span all
+    # inputs/origins (and thus do).  In our desired outcome, the two kinds of
+    # files are comingled.  Without changing the main workflow, the mismatch
+    # has to be reconciled by the upload rule.  Thus, the upload rule enforces
+    # and uses only single-origin configurations.  How third-wave.
+    #   -trs, 7 May 2021
+    if len(config["S3_DST_ORIGINS"]) != 1:
+        raise Exception(f'The "upload" rule requires a single value in S3_DST_ORIGINS (got {config["S3_DST_ORIGINS"]!r}).')
+
+    origin = config["S3_DST_ORIGINS"][0]
```

Comment:
An alternative approach would be to hardcode that this rule only works for profiles defining a single input, which must be either `gisaid` or `open`. This would allow us to modify which files we upload, potentially starting with a small set of files for `open`. 

(Yes, I realise that I was the one who added the `S3_DST_ORIGINS` key in the first place.)